```
some/path/lib/DxilValidation/DxilValidation.cpp:3314:24: error: no member named 'getOperand' in 'llvm::Value'
 3314 |             Ptr = Ptr->getOperand(0);
```

Looks like that change doesn't work because `Value` doesn't have a `getOperand`. I think casting to `Instruction` here should be good enough?

--- a/lib/DxilValidation/DxilValidation.cpp
+++ b/lib/DxilValidation/DxilValidation.cpp
@@ -3309,12 +3309,22 @@ static void ValidateFunctionBody(Function *F, ValidationContext &ValCtx) {
           // comes from a global variable. This was unncessary before SM 6.9
           // because everything was scalarized, but now we can have arrays of
           // vectors in TGSM, so we need to allow GEPs and bitcasts.
-          Value *Ptr = &I;
-          while (Ptr && (isa<GEPOperator>(Ptr) || isa<BitCastOperator>(Ptr)))
-            Ptr = Ptr->getOperand(0);
-          if (!isa<GlobalVariable>(Ptr)) {
-            ValCtx.EmitInstrError(
-                &I, ValidationRule::InstrFailToResloveTGSMPointer);
+          if (isa<GetElementPtrInst>(&I) || isa<BitCastInst>(&I)) {
+            Value *Ptr = cast<Instruction>(&I)->getOperand(0);
+            while (Ptr) {
+              if (GEPOperator *GEP = dyn_cast<GEPOperator>(Ptr)) {
+                Ptr = GEP->getPointerOperand();
+                continue;
+              }
+              if (BitCastOperator *BC = dyn_cast<BitCastOperator>(Ptr)) {
+                Ptr = BC->getOperand(0);
+                continue;
+              }
+              break;
+            }
+            if (!isa<GlobalVariable>(Ptr)) {
+              ValCtx.EmitInstrError(
+                  &I, ValidationRule::InstrFailToResloveTGSMPointer);
             }
           } else {
             ValCtx.EmitInstrError(

--- a/lib/DxilValidation/DxilValidation.cpp
+++ b/lib/DxilValidation/DxilValidation.cpp
@@ -3306,7 +3306,7 @@ static void ValidateFunctionBody(Function *F, ValidationContext &ValCtx) {
       if (PointerType *PT = dyn_cast<PointerType>(I.getType())) {
         if (PT->getAddressSpace() == DXIL::kTGSMAddrSpace) {
           // Walk through GEPs and bitcasts to ensure the pointer ultimately
-          // comes from a global variable. This was unncessary before SM 6.9
+          // comes from a global variable. This was unnecessary before SM 6.9
           // because everything was scalarized, but now we can have arrays of
           // vectors in TGSM, so we need to allow GEPs and bitcasts.
           if (isa<GetElementPtrInst>(&I) || isa<BitCastInst>(&I)) {

